overlord/snapstate/snapstate.go

```diff
@@ -206,6 +206,13 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 		prev = unlink
 	}
 
+	if !release.OnClassic && snapsup.Type == snap.TypeGadget {
+		// XXX: gadget update currently for core systems only
+		gadgetUpdate := st.NewTask("update-gadget", fmt.Sprintf(i18n.G("Update gadget %q%s"), snapsup.InstanceName(), revisionStr))
```

Comment:
should we call the task actually  "update-gadget-assets" ?

the summary should also be  `Update assets from gadget "foo" (12)` perhaps?

Reply:
Right, that sounds like a good idea and matches what the task does.

Reply:
also should we do this only if snapst.IsInstalled ? it will always be a no-op in the other case, no?

Reply:
Task handler does the check as well, As discussed, we'll check in both places.

Reply:
Updated #6750 with the new name as well.